src/cplus_plugin/main.py

```diff
@@ -422,17 +425,6 @@ def initialize_model_settings():
             imp_model = settings_manager.get_implementation_model(imp_model_uuid)
             if imp_model is None:
                 settings_manager.save_implementation_model(imp_model_dict)
-            else:
```

Comment:
@gkahiu this step contained logic that worked together with the initialization functionality to ensure the models are correctly loaded up, aren't we going to have regressions by removing it?

Reply:
Thanks @Samweli for raising this. It is something that I also pondered on - for development versions then there is a risk of regression because of the various iterations that the `ImplementationModel` model has undergone hence not all attributes might be updated. However, being that its a small group of us who are currently developing and testing then the risk is quite low. Also, most of us (who are testing) already have the latest version of the model hence there might be no regressions. If running the plugin for the first time, then there is no risk whatsoever since the IM data model version will be up-to-date.

The primary reason I removed the updating part of the code was because it was causing issues when a user modifies some attributes of a default IM. On restarting QGIS or reloading the plugin, the code would revert back to the default IM values defined in `defaults.py`. While we could still update the values for the attributes, the dilemma was which ones do we use to update the IM i.e. the default value or the (updated) user-defined value? Hence, I went with the safer option of removing that part of the code. 

However, am happy to discuss further and explore additional options of managing this.